students/Harry_Maher/Session04/_filepractice.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+"""
+    Write a program which prints the full path for all files in the current directory, one per line
+    Write a program which copies a file from a source, to a destination (without using shutil, or the OS copy command).
+        Advanced: make it work for any size file: i.e. don’t read the entire contents of the file into memory at once.
+        This should work for any kind of file, so you need to open the files in binary mode: open(filename, 'rb') (or 'wb' for writing). Note that for binary files, you can’t use readline() – lines don’t have any meaning for binary files.
+        Test it with both text and binary files (maybe jpeg or something of your choosing).
+"""
+import os
+
+def print_all_files():
+    for file in os.listdir():
+        print(os.getcwd()+"/"+file) #though windows uses "\" for some reason sometimes?
+ 
+print_all_files()
+
+
+def copy_file(file,output_file):
+    with open(file, 'rb') as infile, open(output_file, 'wb') as outfile:
+        outfile.write(infile.read())
+
+copy_file("copythis.txt", "output_this.txt")
+
+
+"""
+
+Write a little script that reads that students.txt, and generates a list of all the languages that have been used.
+
+What might be the best data structure to use to keep track of bunch of values without duplication?
+Extra challenge: keep track of how many students specified each language
+
+"""
+
+with open("students.txt") as f:
+    students = f.read().splitlines()
```

Comment:
How would this work differently if you used readlines() instead of read()?

Reply:
I guess I wanted to get rid of the "\n" char
https://stackoverflow.com/questions/15233340/getting-rid-of-n-when-using-readlines